### ISSUE TITLE
New buzzwords 5 Essential Pieces of the Deep Learning Puzzle

### ISSUE THREAD

**Issue by sudocurse** [sudocurse/reading-list#4511]
5 Essential Pieces of the Deep Learning Puzzle<br>
http://ift.tt/2DKfgK1<br>
<img width="640" height="431" src="https://storage.googleapis.com/cdn.thenewstack.io/media/2018/01/5585d863-puzzle-1538246_640.png" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2018/01/5585d863-puzzle-1538246_640.png">
<p>Deep learning is quickly becoming one of the most sought-after applications of computer science, and it&rsquo;s no wonder why. Deep learning allows computers to learn and make decisions based on massive datasets. It enables companies to attack otherwise impossible problems in domains like speech recognition, computer vision, natural language processing, and more.</p>
<p>Yet many companies struggle to solve the <a href="https://www.thenewstack.io/tag/Deep-Learning" class="local-link">deep learning puzzle</a> and to capitalize on its full potential. It can be expensive to build and maintain, it requires specialized expertise, and it is still a fairly new concept to the private sector. As companies start to explore this powerful technology there are different ways to make sure they get the most out of it.</p>
<p dir="ltr">For companies who want to implement deep learning, there are five pieces they must have in place to ensure they are making the most of it.</p>
<h3>1: A Way to Quantify Success</h3>
<a href="https://www.sigopt.com" class="clearfix infoBlock" target="_blank"></a>
<div class="infoBlockImageBlock"><img class="infoBlockImage" src="https://storage.googleapis.com/cdn.thenewstack.io/media/2017/11/16d5cbe2-scott-clark.jpg" style="border-radius: 50%;"></div>
<div class="infoBlockTextBlock">
<div class="infoBlockTitle">Scott Clark, co-founder and CEO of SigOpt</div>
<div class="infoBlockText">Scott Clark is the co-founder and CEO of SigOpt. He has been applying optimal learning techniques in industry and academia for years, from bioinformatics to production advertising systems. Before SigOpt, Scott worked on the ad targeting team at Yelp leading the charge on academic research and outreach with projects like the Yelp Dataset Challenge and open sourcing MOE. Scott holds a PhD in Applied Mathematics and an MS in Computer Science from Cornell University and BS degrees in Mathematics, Physics, and Computational Physics from Oregon State University. Scott was chosen as one of Forbes&rsquo; 30 under 30 in 2016.</div>
</div>
<p>First, before any modeling efforts are started &mdash; whether they are in deep learning, more traditional machine learning, or data science in general &mdash; it is important to quantify a measure of success. This process involves determining the metrics that these models and efforts will track and be judged by. Ideally, there is a way to trace the performance of a model directly back to business value. As a fraud detection model gets more accurate you may lose less money to fraud. As a recommendation model gets better you may sell more products. Sometimes this is more difficult when there are many steps between a specific model and explicit business value, but it is an extremely important step.</p>
<blockquote>
<p>Ideally, there is a way to trace the performance of a model directly back to business value.</p>
</blockquote>
<p>Without a metric tied to business value, it is impossible to judge whether you are making progress or if the model itself has any value at all. Even in circumstances where it seems straightforward, this can be difficult. In the case of fraud, it is important to make sure the definition of accuracy matches what the business needs. Catching 99 percent of fraud is good, but if all of the 1 percent that slips through is $10 million or more then it is very bad for the business. This measurement of success is always incredibly domain specific and will be different for every business and application.</p>
<p>Not defining these metrics or using standard academic measures can lead to wasted time, lowered business outcomes, and models that are built with no real purpose. Without the ability to measure performance of these methods and relate that to business value business can waste a ton of time and money to effectively build very expensive algorithmic toys.</p>
<h3>2: An Abundance of Clean Data</h3>
<p>The amount of applications and Internet of Things (IoT) devices in the enterprise means there&rsquo;s no shortage of data. But many datasets can&rsquo;t be used for deep learning because they&rsquo;re too small, not structured, or untrustworthy. If machine learning models are trained with incomplete, unstructured, or inaccurate data, they can&rsquo;t arrive at the best outcomes at best, and can even actively hurt the underlying application they are being applied to.</p>
<p>For example, imagine you&rsquo;re training a <a href="https://thenewstack.io/tag/neural-networks" class="local-link">neural network</a> to classify images between dogs and cats. To do that, the neural network must be trained on many images that are labeled with the corresponding right answer. By feeding the machine enough images of different types of dogs and cats, the neural network will eventually be able to learn patterns to accurately identify the differences. If you only have a few images, only a single breed is included, or if the data is mislabeled the resulting model will not achieve the intended general purpose. Companies must ensure they have enough data that is clean and clearly labeled to improve the accuracy of their deep learning models. This step must be achieved before any training or learning is possible.</p>
<h3>3: A Deep Learning Framework</h3>
<p>Deep learning frameworks are the libraries and programming models upon which developers build deep learning applications. Frameworks are necessary for effective deep learning &mdash; without them, companies can&rsquo;t do things like image and speech recognition or language understanding. However, it&rsquo;s expensive and time-consuming to build these frameworks from scratch.</p>
<p>Luckily, tools like <a href="https://www.tensorflow.org/" class="ext-link" rel="external nofollow">Tensorflow</a>, <a href="https://mxnet.apache.org/" class="ext-link" rel="external nofollow">MXNet</a> and <a href="https://caffe2.ai/" class="ext-link" rel="external nofollow">Caffe2</a> provide ready-to-use, cloud-based deep learning frameworks that are easy to implement. The frameworks come with open source examples, like natural language processing systems and image recognition systems. If a company has the data to train the model, it&rsquo;s fairly simple to plug in the data without having to know how the underlying framework was built. When choosing a deep learning framework, companies should consider factors like available resources for development issues, ease of use and deployment, and customization. This step must be achieved before you can build your first model.</p>
<h3>4: A Robust Cloud Infrastructure</h3>
<p>Deep learning requires extreme computational resources. Traditional IT infrastructure is inadequate for deep learning because it can&rsquo;t process large volumes of data required to drive insights. GPU-enabled cloud infrastructure on platforms like Amazon Web Services and Microsoft Azure has made it easier than ever for companies to build and scale deep learning pipelines. GPU-based infrastructure can offload the massively parallel, compute-intensive portions of an application to the GPU, while the remainder of the code remains on the CPU, enabling applications to run <a href="https://aws.amazon.com/blogs/ai/fast-cnn-tuning-with-aws-gpu-instances-and-sigopt/" target="_blank" data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=https://aws.amazon.com/blogs/ai/fast-cnn-tuning-with-aws-gpu-instances-and-sigopt/&amp;source=gmail&amp;ust=1516998844799000&amp;usg=AFQjCNGK-XjntxFVUOEIUR-px63nzlicGA" class="ext-link" rel="external nofollow">up to 50 times faster and more effectively in practice</a>. This step must be achieved before you can scale your model to production level datasets and enterprise applications.</p>
<h3>5: A Way to Optimize the Models</h3>
<p>Once you have the data, the frameworks are in place, and the hardware infrastructure is robust, the final step keeping deep learning performance from achieving its true potential is model optimization and hyperparameter tuning. For any deep learning model, researchers need to set dozens of hyperparameters before they can input data to train the model. These parameters include things like the number of layers in the network, the number of nodes per layer, or the learning rate. If the initial training session performed poorly, researchers will often tune a few knobs and give it another go or they may rely on expensive, brute force methods like <a href="http://nogridsearch.com/" target="_blank" data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://nogridsearch.com/&amp;source=gmail&amp;ust=1516998844799000&amp;usg=AFQjCNH91uFW_PoUvAaoWAne3_k5At_nvg" class="ext-link" rel="external nofollow">grid search</a>. These approaches to tuning deep learning models are time intensive and offer no guarantee of good results.</p>
<p>To make this easier on developers, data scientists and researchers, software-as-a-service solutions like <a href="http://sigopt.com/" target="_blank" data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://sigopt.com/&amp;source=gmail&amp;ust=1516998844799000&amp;usg=AFQjCNEylya7SGUGo894UPq0k0DCKQJpBA" class="ext-link" rel="external nofollow">SigOpt</a>, <a href="https://aws.amazon.com/blogs/aws/sagemaker/" target="_blank" data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=https://aws.amazon.com/blogs/aws/sagemaker/&amp;source=gmail&amp;ust=1516998844799000&amp;usg=AFQjCNHZkeKbO8HfL_c1zsBN6ILNRXoG5g" class="ext-link" rel="external nofollow">Amazon SageMaker</a>, and <a href="https://cloud.google.com/ml-engine/docs/using-hyperparameter-tuning" target="_blank" data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=https://cloud.google.com/ml-engine/docs/using-hyperparameter-tuning&amp;source=gmail&amp;ust=1516998844799000&amp;usg=AFQjCNFgSJQLB2STXby4TSW6lNVxikRRYQ" class="ext-link" rel="external nofollow">Google Hypertune</a> exist to take the guesswork out of hyperparameter tuning. These optimization tools sit on top of models and observe their inputs and outputs to help researchers identify where they need to adjust their systems. Optimizing machine learning algorithms and deep learning frameworks can provide many orders of magnitude improvements in <a href="https://devblogs.nvidia.com/parallelforall/sigopt-deep-learning-hyperparameter-optimization/" target="_blank" data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=https://devblogs.nvidia.com/parallelforall/sigopt-deep-learning-hyperparameter-optimization/&amp;source=gmail&amp;ust=1516998844799000&amp;usg=AFQjCNE1VeQmBnj8ijBBKGhcjmqE6_u4Nw" class="ext-link" rel="external nofollow">accuracy, performance and efficiency</a>. This step allows you to get the most business value out of your deep learning investment.</p>
<h3>Take These Steps</h3>
<p>Deep learning is still gaining a foothold among large companies and it can be puzzling to know where to start. There are many tools available to ensure companies can capitalize on all deep learning has to offer. The technology will continue to transform how businesses operate, and as it does, companies will need to have these pieces in place to ensure they don&rsquo;t get swept away by the competition.</p>
<p>By following these steps companies can make sure their models are aimed in the right direction, and their deep learning efforts are possible, scalable, and optimal.</p>
<p class="attribution">Feature image <a href="https://pixabay.com/en/puzzle-isolated-transparent-plan-1538246/" class="ext-link" rel="external nofollow">via</a> Pixabay.</p>
<p>The post <a rel="nofollow" href="https://thenewstack.io/five-essential-pieces-deep-learning-puzzle/">5 Essential Pieces of the Deep Learning Puzzle</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br>
<br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
January 29, 2018 at 12:00PM
